modules/extension/wms/src/main/java/org/geotools/ows/wms/map/WMSCoverageReader.java

```diff
@@ -207,6 +212,31 @@ void addLayer(Layer layer, String style) {
         updateBounds();
     }
 
+    /** Returns the first CRS from layer srs list that matches with bounding boxes map keys. */
+    private String getPriorityCRS(Layer layer) {
+        List<String> priorityCRSs = getPriorityCRSs(layer);
+        if (!priorityCRSs.isEmpty()) {
+            String srs = priorityCRSs.get(0);
+            try {
+                // check it's valid, if not we crap out and move to the next
+                CRS.decode(srs);
+                return srs;
+            } catch (Exception e) {
+                // it's fine, we could not decode that code
+            }
+        }
+        return null;
+    }
+
+    private List<String> getPriorityCRSs(Layer layer) {
```

Comment:
Naming is vague, "priority" might mean many things. What this does, is to return the CRSs with a matching bounding box definition. ``getCRSWithBoundingBoxes`` maybe?

Reply:
Right, better name, fixed.